tests/queries/0_stateless/02156_minus_op_with_datatime64.sql

```diff
@@ -0,0 +1,53 @@
+-- Test subtraction without materialize()
+SELECT
+-- DateTime64 (scale 3) - DateTime
+toDateTime64('2023-10-01 12:00:00', 3) - toDateTime('2023-10-01 11:00:00') AS result_no_materialize_1,
+toDateTime64('2023-10-01 12:00:00.123', 3) - toDateTime('2023-10-01 11:00:00') AS result_no_materialize_2,
+
+-- DateTime64 (scale 0) - DateTime
+toDateTime64('2023-10-02 12:00:00', 0) - toDateTime('2023-10-01 11:00:00') AS result_no_materialize_3,
+
+-- DateTime64 (scale 3) - DateTime64 (scale 6)
+toDateTime64('2023-10-01 12:00:00', 3) - toDateTime64('2023-10-01 11:00:00', 6) AS result_no_materialize_4,
+
+-- DateTime64 (scale 6) - DateTime64 (scale 3)
+toDateTime64('2023-10-01 12:00:00', 6) - toDateTime64('2023-10-01 11:00:00', 3) AS result_no_materialize_5,
```

Comment:
It would be nice to add a fraction part here, like `toDateTime64('2023-10-01 12:00:00.123456', 6) - toDateTime64('2023-10-01 11:00:00.123', 3)`

Reply:
Done